sw/airborne/modules/imu/imu.c

```diff
@@ -776,9 +821,16 @@ static void imu_accel_raw_cb(uint8_t sender_id, uint32_t stamp, struct Int32Vect
 
   // Scale the accel
   struct Int32Vect3 scaled, scaled_rot;
-  scaled.x = (accel->unscaled.x - accel->neutral.x) * accel->scale[0].x / accel->scale[1].x;
-  scaled.y = (accel->unscaled.y - accel->neutral.y) * accel->scale[0].y / accel->scale[1].y;
-  scaled.z = (accel->unscaled.z - accel->neutral.z) * accel->scale[0].z / accel->scale[1].z;
+  if(accel->calibrated.scale && !accel->calibrated.scale_f) {
```

Comment:
Can't we just calculate the `scale_f` during the initialization if the `scale` is set? Then we don't have this if case everywhere.